test/test_eos_helmholtz.cpp

```diff
@@ -0,0 +1,170 @@
+//------------------------------------------------------------------------------
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifdef SINGULARITY_TEST_HELMHOLTZ
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+
+#include <array>
+#include <limits>
+#include <string>
+
+#ifndef CATCH_CONFIG_RUNNER
+#include "catch2/catch.hpp"
+#endif
+
+#include <ports-of-call/portability.hpp>
+#include <ports-of-call/portable_arrays.hpp>
+#include <ports-of-call/portable_errors.hpp>
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/eos/eos.hpp>
+#include <test/eos_unit_test_helpers.hpp>
+
+using Catch::Matchers::WithinRel;
+using singularity::Helmholtz;
+const std::string filename = "../test/helmholtz/helm_table.dat";
+SCENARIO("Helmholtz equation of state - Table interpolation (tgiven)", "[HelmholtzEOS]") {
```

Comment:
Looks like the tests are not GPU. We should test on device before merge. It's an easy change, and I can add it.

Reply:
I included a test on device, not sure if that is what you meant. In the second test I also tired define the input values as ``Kokkos::View`` in the same way as in the first test, but for some reason the test then fails. (I made sure to use the host values for the comparison, but the ``temp_new`` value reported by the ``REQUIRE_THAT`` call differs from the value to match. However, if I print the ``temp_new`` value before that, it matches the compared value perfectly. I presume it is a trivial thing I just missed, but I wasn't able to find the reason for that so far.

In any case, the current tests should test on device. Tested locally on a RTX 4060 with CUDA 12.2.91. + gcc 12.3.0

Reply:
I'm not sure why either test would work---you weren't testing on device, because you had normal for loops for the test code, not `portableFor` loops. I've fixed it, and tests now run on device.

Reply:
Thanks for fixing that! I've included a fix to make the tests pass when not using Kokkos. If you could double check that, I'm not sure if I understood your intention for ``nwrong = 1`` correct. You set it to one such that in case the reduction fails the test fails, right? I'm not sure how the ``portableReduce`` behaves without Kokkos, but I assume it does not do anything in that case, i.e. there is no danger of setting ``nwrong = 0`` initially?

Reply:
No there's no danger---Kokkos sets it to zero before the reduction but I'm not sure that the cpu version does. Good fix setting it to zero initially